### PR TITLE
KUZ-412 : Profile creation was not stopped by role that don't exist

### PR DESCRIPTION
**Pull request by dbengsch**

### CODE REVIEW

File: lib/api/core/models/repositories/profileRepository.js [hunk 1]
```diff
@@ -130,7 +133,8 @@ module.exports = function (kuzzle) {
         // Fail if not all roles are found
         var rolesNotFound = _.difference(data.roles, extractRoleIds(roles));
         if (rolesNotFound.length) {
-          deferred.reject(new NotFoundError('Unable to hydrate profile with roles'));
+          deferred.reject(new NotFoundError('Unable to hydrate profile with roles: some roles don\'t exist.'));
```
Comment (scottinet) [inline lib/api/core/models/repositories/profileRepository.js]: Nitpicking: to make the error message more explicit, we could add the non-existing roles. 
Something like this:
```js
new NotFoundError(`Unable to create profile ${requestObject.data._id}. The following roles don't exist: ${rolesNotFound}`)
```

**Comment by ballinette** [comment]
+1

File: lib/api/core/models/repositories/profileRepository.js [hunk 2]
```diff
@@ -130,7 +130,8 @@ module.exports = function (kuzzle) {
         // Fail if not all roles are found
         var rolesNotFound = _.difference(data.roles, extractRoleIds(roles));
         if (rolesNotFound.length) {
-          deferred.reject(new NotFoundError('Unable to hydrate profile with roles'));
+          deferred.reject(new NotFoundError(`Unable to hydrate the profile ${requestObject.data._id}. The following roles don't exist: ${rolesNotFound}`));
```
Comment (ballinette) [inline lib/api/core/models/repositories/profileRepository.js]: requestObject is not defined here....
maybe you wanted to write 

    deferred.reject(new NotFoundError(`Unable to hydrate the profile ${requestObject.data._id}. The following roles don't exist: ${rolesNotFound}`));

?

**Comment by AnthonySendra** [comment]
+1

**Comment by scottinet** [comment]
+1!

**Pull request merged**